src/Hosting/WindowsServices/ref/Microsoft.Extensions.Hosting.WindowsServices.netstandard2.0.cs

```diff
@@ -16,7 +22,7 @@ public static partial class WindowsServiceHelpers
     }
     public partial class WindowsServiceLifetime : System.ServiceProcess.ServiceBase, Microsoft.Extensions.Hosting.IHostLifetime
     {
-        public WindowsServiceLifetime(Microsoft.Extensions.Hosting.IHostEnvironment environment, Microsoft.Extensions.Hosting.IHostApplicationLifetime applicationLifetime, Microsoft.Extensions.Logging.ILoggerFactory loggerFactory, Microsoft.Extensions.Options.IOptions<Microsoft.Extensions.Hosting.HostOptions> optionsAccessor) { }
+        public WindowsServiceLifetime(Microsoft.Extensions.Hosting.IHostEnvironment environment, Microsoft.Extensions.Hosting.IHostApplicationLifetime applicationLifetime, Microsoft.Extensions.Logging.ILoggerFactory loggerFactory, Microsoft.Extensions.Options.IOptions<Microsoft.Extensions.Hosting.HostOptions> hostOptionsAccessor, Microsoft.Extensions.Options.IOptions<Microsoft.Extensions.Hosting.WindowsServiceLifetimeOptions> windowsServiceOptionsAccessor) { }
```

Comment:
We can make this change without making this a breaking change. Just add an additional ctor that adds an `IOptions<WindowsServiceLifetimeOptions>` and have the other ctor pass in null.

Reply:
Also let's leave the first parameter named `optionsAccessor`. Avoids needless confusion.

Reply:
@Kahbazi can you make this change? Basically, leave the existing constructor and add this as a new constructor instead of a replacement. The existing constructor should remain unmodified (even parameter names), to avoid API breaks.

Reply:
@anurse I just did it in https://github.com/aspnet/Extensions/pull/2626/commits/5ce2efc27955ec6d95f607b64156fdd57ed859a9